bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java

```diff
@@ -478,8 +478,6 @@ private String stateToString(State state) {
             value = ((OnOffType) state) == OnOffType.ON ? DIGITAL_VALUE_ON : DIGITAL_VALUE_OFF;
         } else if (state instanceof OpenClosedType) {
             value = ((OpenClosedType) state) == OpenClosedType.OPEN ? DIGITAL_VALUE_ON : DIGITAL_VALUE_OFF;
-        } else if (state instanceof HSBType) {
```

Comment:
Explain or revert this change.

Reply:
`HSBType` is a subclass of `PercentType`, which is in turn a subclass of `DecimalType`. So this check is being made after it has been established that `state` is not an instance of `HSBType`. The change removes the redundant check and the dead code, preserving the semantics of the code. On the other hand, perhaps there is a bug and the code is not doing what was intended. If that is the case, the test for `HSBType` should be done before the test for `DecimalType`. Which solution do you prefer ?

Reply:
I guess we'll leave it as you have it, since that's the way it's been operating in practice.